### ISSUE TITLE
Crash on empty buffer

### ISSUE THREAD

**Issue by psawaya** [barrysteyn/node-scrypt#97]
Hi all,

Thanks for the fantastic library. Just wanted to point out that I'm getting the crash on empty buffer (#70) with the new API.

```
> scrypt.verifyKdf(new Buffer('', 'base64'),new Buffer('foo')).then(function(result) { console.log(result); })
Segmentation fault: 11
```

**Comment by barrysteyn** [comment]
Yeah, that makes sense. Actually (embarrassingly), I knew about this. I will harden the code when I get round to it.

**Comment by barrysteyn** [comment]
Fixed in daed9929a2b87cc71463c20cec641172d39734e3
